src/test/java/echoserver/FakeServerSocket.java

```diff
@@ -6,15 +6,28 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public class FakeServerSocket extends ServerSocket {
-    private final List<FakeClientSocket> fakeClientSockets;
+class FakeServerSocket extends ServerSocket {
+    private List<FakeClientSocket> fakeClientSockets = new ArrayList<>();
+    private final OutputStream output;
 
-    FakeServerSocket(ArrayList<FakeClientSocket> fakeClientSockets) throws IOException {
-        this.fakeClientSockets = fakeClientSockets;
+    FakeServerSocket(OutputStream output) throws IOException {
+        this.output = output;
     }
 
     @Override
     public Socket accept() {
         return fakeClientSockets.remove(0);
-       }
+    }
+
+    void setupFakeClientSockets() {
+        var input1 = new ByteArrayInputStream("Client 1's message".getBytes());
+        var input2 = new ByteArrayInputStream("Client 2's message".getBytes());
+        var input3 = new ByteArrayInputStream("Client 3's message".getBytes());
+        var input4 = new ByteArrayInputStream("Client 4's message".getBytes());
+
+        fakeClientSockets.add(new FakeClientSocket(input1, output));
```

Comment:
Passing the same output to all fakeclients means they will all have the same output - is that what you want?

Reply:
Thanks for spotting this. I changed this in a later commit in, within this pull request